test/main/async.js

```diff
@@ -173,6 +175,17 @@ QUnit.test( "fails if callback is called more than callback call count", functio
 	done();
 } );
 
+QUnit.test( "errors if callback is called after test - part 1", function( assert ) {
+	assert.expect( 0 );
+	globalDone = assert.async();
+	globalDone();
+} );
+
+QUnit.test( "errors if callback is called after test - part 2", function( assert ) {
+	assert.throws( globalDone, /assert.async callback called after test finished./ );
+	globalDone = null;
+} );
+
```

Comment:
There's a small part of me that wonders if we should just have a nested module to allow for a smaller scope of closure, but this looks fine as is.